### PR TITLE
Handle PRs from deleted repositories.

### LINKED ISSUE #721: Maintainer Workflow: assert failure on loading PR from ghost account
Attempting to load a PR such as https://github.com/aspnet/mvc/pull/2057 which is created by a ghost account causes an assertion failure. We also need to disable attempting to check out such a PR.

### PR DESCRIPTION
**Pull request by grokys**
If a PR was submitted from a fork and that repository subsequently deleted, we previously asserted in the PR list. Handle this case and disable the checkout button in the PR details view.

This PR builds on #720 so that should be merged first.

Fixes #721

### CODE REVIEW

**Comment by paladique** [comment]
Gave this a try and worked fine for me! Got the message "The source repository is no longer available." when I moused over the checkout link. I was questioning whether or not this would be obvious to the user, but I imagine they'll navigate to the conversation in the browser and see what happened.

File: src/GitHub.App/ViewModels/PullRequestDetailViewModel.cs [hunk 1]
```diff
@@ -325,9 +325,17 @@ public override void Initialize([AllowNull] ViewWithData data)
...
                     "Checkout to " + (await pullRequestsService.GetDefaultLocalBranchName(repository, Model.Number, Model.Title));
-                var disabled = await pullRequestsService.IsWorkingDirectoryClean(repository) ?
-                    null :
-                    "Cannot checkout as your working directory has uncommitted changes.";
+                var clean = await pullRequestsService.IsWorkingDirectoryClean(repository);
+                string disabled = null;
+
+                if (pullRequest.Head == null || !pullRequest.Head.RepositoryCloneUrl.IsValidUri)
+                {
+                    disabled = "The source repository is no longer available.";
```
Comment (shana) [inline src/GitHub.App/ViewModels/PullRequestDetailViewModel.cs]: These should go into the resources file. Are we doing that on a later pass for all strings, or doing it as we go along? I remember we talked about this but not what we decided.

Comment (paladique) [inline src/GitHub.App/ViewModels/PullRequestDetailViewModel.cs]: Yup! There's an issue to do that after #713 is taken care of: https://github.com/github/VisualStudio/issues/715

Comment (shana) [inline src/GitHub.App/ViewModels/PullRequestDetailViewModel.cs]: Aaaah indeed indeed! :+1:

**Comment by shana** [comment]
> Gave this a try and worked fine for me! Got the message "The source repository is no longer available." when I moused over the checkout link. I was questioning whether or not this would be obvious to the user, but I imagine they'll navigate to the conversation in the browser and see what happened.

That's a good point! Let's keep that in mind while we test and see if it's enough or if we need to add more obvious messaging.

**Comment by shana** [comment]
Linkback to #733 which depends on this.

**Pull request merged**